openstudiocore/src/energyplus/ReverseTranslator/ReverseTranslateEnergyManagementSystemMeteredOutputVariable.cpp

```diff
@@ -0,0 +1,159 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2016, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#include "../ReverseTranslator.hpp"
+
+#include "../../model/EnergyManagementSystemMeteredOutputVariable.hpp"
+#include "../../model/EnergyManagementSystemMeteredOutputVariable_Impl.hpp"
+#include "../../model/EnergyManagementSystemProgram.hpp"
+#include "../../model/EnergyManagementSystemProgram_Impl.hpp"
+#include "../../model/EnergyManagementSystemSubroutine.hpp"
+#include "../../model/EnergyManagementSystemSubroutine_Impl.hpp"
+#include "../../model/EnergyManagementSystemGlobalVariable.hpp"
+#include "../../model/EnergyManagementSystemGlobalVariable_Impl.hpp"
+
+#include <utilities/idd/EnergyManagementSystem_MeteredOutputVariable_FieldEnums.hxx>
+#include "../../utilities/idd/IddEnums.hpp"
+#include <utilities/idd/IddEnums.hxx>
+
+using namespace openstudio::model;
+
+namespace openstudio {
+
+namespace energyplus {
+
+OptionalModelObject ReverseTranslator::translateEnergyManagementSystemMeteredOutputVariable(const WorkspaceObject & workspaceObject)
+{
+  if (workspaceObject.iddObject().type() != IddObjectType::EnergyManagementSystem_MeteredOutputVariable) {
+    LOG(Error, "WorkspaceObject is not IddObjectType: EnergyManagementSystem_MeteredOutputVariable");
+    return boost::none;
+  }
+
+  //make sure all other objects are translated first except below
+  for (const WorkspaceObject& workspaceObject : m_workspace.objects()) {
+    if ((workspaceObject.iddObject().type() != IddObjectType::EnergyManagementSystem_Program)
+      && (workspaceObject.iddObject().type() != IddObjectType::EnergyManagementSystem_Subroutine)
+      && (workspaceObject.iddObject().type() != IddObjectType::EnergyManagementSystem_ProgramCallingManager)
+      && (workspaceObject.iddObject().type() != IddObjectType::EnergyManagementSystem_MeteredOutputVariable)
+      && (workspaceObject.iddObject().type() != IddObjectType::EnergyManagementSystem_OutputVariable)) {
+      translateAndMapWorkspaceObject(workspaceObject);
```

Comment:
yeah, kinda stinks but i needed a way to enforce translation order.  Since RT isnt really encouraged anymore I thought the penalty might be acceptable.